packages/starlight/__e2e__/ssr.test.ts

```diff
@@ -0,0 +1,68 @@
+import { expect, testFactory } from './test-utils';
+import assert from 'node:assert';
+import * as cheerio from 'cheerio';
+
+const test = testFactory('./fixtures/ssr/');
+
+test.beforeEach(() => {
+	delete process.env.STARLIGHT_PRERENDER;
+});
+
+test('Render page on the server', async ({ page, getProdServer }) => {
+	const starlight = await getProdServer();
+	await starlight.goto('/demo');
+
+	await expect(page.locator('#server-check')).toHaveText('On server');
+});
+
+test('Render 404 page on the server', async ({ page, getProdServer }) => {
+	const starlight = await getProdServer();
+	await starlight.goto('/not-found');
+
+	await expect(page.locator('#server-check')).toHaveText('On server');
+});
+
+test('SSR mode renders the same content page as prerendering', async ({
+	getProdServer,
+	makeServer,
+}) => {
+	const starlight = await getProdServer();
+	const ssrContent = await starlight.goto('/content').then((res) => res?.text());
+	assert(ssrContent);
+
+	process.env.STARLIGHT_PRERENDER = 'yes';
+	const prerenderStarlight = await makeServer('prerender');
+	const prerenderContent = await prerenderStarlight.goto('/content').then((res) => res?.text());
+	assert(prerenderContent);
+
+	expectEquivalentHTML(prerenderContent, ssrContent);
+});
+
+test('SSR mode renders the same splash page as prerendering', async ({
+	getProdServer,
+	makeServer,
+}) => {
+	const starlight = await getProdServer();
+	const ssrContent = await starlight.goto('/').then((res) => res?.text());
+	assert(ssrContent);
+
+	process.env.STARLIGHT_PRERENDER = 'yes';
+	const prerenderStarlight = await makeServer('prerender');
+	const prerenderContent = await prerenderStarlight.goto('/').then((res) => res?.text());
+	assert(prerenderContent);
+
+	expectEquivalentHTML(prerenderContent, ssrContent);
+});
+
+function expectEquivalentHTML(a: string, b: string) {
+	const a$ = cheerio.load(a);
+	const b$ = cheerio.load(b);
+
+	a$('script[src]').attr('src', '');
+	a$('link[href]').attr('href', '');
+
+	b$('script[src]').attr('src', '');
+	b$('link[href]').attr('href', '');
+
+	expect(a$.html()).toEqual(b$.html());
+}
```

Comment:
Wonder if we could use something lighter than Cheerio for this. I quite like `linkedom` and in a quick test it worked just as well and is about half the size ([2.3 MB](https://pkg-size.dev/linkedom) vs [5.1 MB](https://pkg-size.dev/cheerio)).

It’s also nice because its API is the standard DOM API, instead of using cryptic `.attr()` calls:
```diff
- import * as cheerio from 'cheerio';
+ import { parseHTML } from 'linkedom';
```
```suggestion
function expectEquivalentHTML(a: string, b: string) {
	expect(getNormalizedHTML(a)).toEqual(getNormalizedHTML(b));
}

function getNormalizedHTML(html: string) {
	const window = parseHTML(html);
	window.document.querySelectorAll('script[src]').forEach((el) => el.setAttribute('src', ''));
	window.document.querySelectorAll('link[href]').forEach((el) => el.setAttribute('href', ''));
	return window.toString();
}
```
What do you think?

Reply:
I've never heard of linkedom, but if works that's fine by me.

Reply:
It’s a really nice and fast alternative to JSDOM built with linked lists — https://github.com/WebReflection/linkedom